cmd/controller/controller.go

```diff
@@ -134,11 +134,17 @@ func joinController(tokenArg string, certRootDir string) (*token.JoinClient, err
 		return nil, fmt.Errorf("failed to create join client: %w", err)
 	}
 
-	caData, err := joinClient.GetCA()
+	var caData v1beta1.CaResponse
+	err = retry.Do(func() error {
+		caData, err = joinClient.GetCA()
+		if err != nil {
+			return fmt.Errorf("failed to sync CA: %w", err)
+		}
+		return nil
+	})
```

Comment:
FYI, total retry time is ~52 seconds, pause between two last attempts is 25 seconds.

Maybe could be improved by giving up early in case of some unrecoverable failures:

```go
		err = retry.Do(
			func() error {
				....
			},
			retry.RetryIf(
				func(err error) bool {
					return !strings.Contains(err.Error(), "something")
				},
			),
		)
```

Reply:
> Maybe could be improved by giving up early in case of some unrecoverable failures

I actually did think of this, but it's just super hard to make difference what is un-recoverable and what is not. E.g. with k0sctl (as it's super fast) the initial trys might give plain IO timeout (the join API is not yet up on the first controller), some RBAC related 40x (initial controller not yet pushed RBAC stuff for join tokens) or something else.

> FYI, total retry time is ~52 seconds, pause between two last attempts is 25 seconds.
I think this is OK for this case.